File: app/src/main/java/com/sabanciuniv/todoapp/model/FragmentTodoTabs.java
package com.sabanciuniv.todoapp.model;

import android.app.Application;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.sabanciuniv.todoapp.AddNewTodo;
import com.sabanciuniv.todoapp.MainActivity;
import com.sabanciuniv.todoapp.R;
import com.sabanciuniv.todoapp.ToDoApplication;
import com.sabanciuniv.todoapp.TodoRecViewAdapter;
import com.sabanciuniv.todoapp.databinding.FragmentTodoTabsBinding;

import java.util.List;


public class FragmentTodoTabs extends Fragment  {

    private FragmentTodoTabsBinding binding;

    ToDoRepository repo = new ToDoRepository();

    String isChecked;
    public FragmentTodoTabs(String isChecked) {
        this.isChecked = isChecked;
    }

    private final TodoRecViewAdapter.CheckListener checkListener = new TodoRecViewAdapter.CheckListener() {
        @Override
        public void onButtonClicked(String id) {
            Application app = requireActivity().getApplication();
            repo.changeChecked(((ToDoApplication) app).srv, id);
        }
    };
    Handler dataHandler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(@NonNull Message msg) {
            List<ToDo> todoList = (List<ToDo>) msg.obj;
            TodoRecViewAdapter todoRecViewAdapter = new TodoRecViewAdapter(todoList, getContext(), checkListener);
            binding.recViewTodos.setAdapter(todoRecViewAdapter);
            binding.prgBarTodos.setVisibility(View.INVISIBLE);
            binding.recViewTodos.setVisibility(View.VISIBLE);
            return true;
        }
    });

    @Override
    public void onCreate(Bundle savedInstanceState) {super.onCreate(savedInstanceState);}

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentTodoTabsBinding.inflate(inflater, container, false);
        View v =  binding.getRoot();
        binding.recViewTodos.setLayoutManager(new LinearLayoutManager(getContext()));
        binding.prgBarTodos.setVisibility(View.VISIBLE);
        binding.recViewTodos.setVisibility(View.INVISIBLE);
        Application app = requireActivity().getApplication();
        repo.getAllToDos(((ToDoApplication) app).srv,dataHandler);
        return v;
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.swipeRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                binding.prgBarTodos.setVisibility(View.VISIBLE);
                binding.swipeRefresh.setRefreshing(false);
                repo.getAllToDos(((ToDoApplication)requireActivity().getApplication()).srv,dataHandler);
            }
        });

        binding.floatingActionButton.setOnClickListener(v -> {
            Intent i = new Intent(getActivity(), AddNewTodo.class);
            startActivity(i);
        });
    }
}

File: app/src/main/java/com/sabanciuniv/todoapp/fragment/FragmentTodoTabs.kt
package com.sabanciuniv.todoapp.activity

import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.sabanciuniv.todoapp.ToDoApplication
import com.sabanciuniv.todoapp.TodoRecViewAdapter
import com.sabanciuniv.todoapp.databinding.FragmentTodoTabsBinding
import com.sabanciuniv.todoapp.`interface`.RecyclerViewInterface
import com.sabanciuniv.todoapp.model.ToDo
import com.sabanciuniv.todoapp.repository.ToDoRepository

class FragmentTodoTabs(var isChecked: String) : Fragment(), RecyclerViewInterface {

    private var binding: FragmentTodoTabsBinding? = null

    var repo = ToDoRepository()

    lateinit var todoList: MutableList<ToDo>
    lateinit var todoRecViewAdapter : TodoRecViewAdapter





    var dataHandler = Handler { msg ->
        todoList = msg.obj as MutableList<ToDo>
         todoRecViewAdapter = context?.let {
             TodoRecViewAdapter(
                 todoList,
                 it, this
             )
         }!!
        binding!!.recViewTodos.adapter = todoRecViewAdapter
        binding!!.prgBarTodos.visibility = View.INVISIBLE
        binding!!.recViewTodos.visibility = View.VISIBLE
        true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentTodoTabsBinding.inflate(inflater, container, false)
        val v: View = binding!!.root
        binding!!.recViewTodos.layoutManager = LinearLayoutManager(context)
        binding!!.prgBarTodos.visibility = View.VISIBLE
        binding!!.recViewTodos.visibility = View.INVISIBLE
        val app = requireActivity().application
        repo.getAllToDos((app as ToDoApplication).srv, dataHandler, isChecked)
        return v
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val helper = ItemTouchHelper(object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT ) {
            override fun onMove(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder,
                target: RecyclerView.ViewHolder
            ): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
                todoList.removeAt(viewHolder.adapterPosition)
                todoRecViewAdapter.notifyItemRemoved(viewHolder.adapterPosition)
            }
        })
        helper.attachToRecyclerView(binding!!.recViewTodos)
        binding!!.swipeRefresh.setOnRefreshListener {
            binding!!.prgBarTodos.visibility = View.VISIBLE
            binding!!.swipeRefresh.isRefreshing = false
            repo.getAllToDos((requireActivity().application as ToDoApplication).srv, dataHandler, isChecked)
        }
    }

    override fun onCheckboxClicked(position: Int, id: String) {
        val app = requireActivity().application
        todoList.removeAt(position)
        todoRecViewAdapter.notifyItemRemoved(position)
        repo.changeChecked((app as ToDoApplication).srv, id)
    }


}